src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,41 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    return tag != null && !tag.isEmpty() ;
+  }
+
+  private static String extractHashTag(String key, boolean includeBrackets, boolean returnNullWhenTagNotFound) {
```

Comment:
@marcosnils  though I don't really understand your objection to a method containing 2 boolean arguments ( especially as I made an effort to give them clear names ) , I went ahead and omitted the 'includeBrackets' argument as it wasn't really necessary due to the various changes done as part of the pull request :+1:  :smile_cat: